File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class ZoneDemandSupplyItem(BaseModel):
    zone: str
    demand_count: int
    supply_count: int
    shortage: int


class ClusterDistributionItem(BaseModel):
    cluster_label: str
    count: int


class WaitingListItem(BaseModel):
    cluster_label: str
    pending_requests: int


class HeatmapItem(BaseModel):
    zone: str
    demand_intensity: int
    supply_intensity: int


class AnalyticsOverview(BaseModel):
    demand_supply: list[ZoneDemandSupplyItem]
    parent_clusters: list[ClusterDistributionItem]
    childminder_clusters: list[ClusterDistributionItem]
    waiting_list: list[WaitingListItem]
    heatmap: list[HeatmapItem]

File: frontend/src/types/analytics.ts
export type Totals = {
  total_parents: number;
  total_childminders: number;
  total_requests: number;
  total_matches: number;
};

export type ZoneDemandSupplyItem = {
  zone: string;
  demand_count: number;
  supply_count: number;
  shortage: number;
};

export type ClusterDistributionItem = {
  cluster_label: string;
  count: number;
};

export type WaitingListItem = {
  cluster_label: string;
  pending_requests: number;
};

export type HeatmapItem = {
  zone: string;
  demand_intensity: number;
  supply_intensity: number;
};

export type AnalyticsOverview = {
  totals: Totals;
  demand_supply: ZoneDemandSupplyItem[];
  parent_clusters: ClusterDistributionItem[];
  childminder_clusters: ClusterDistributionItem[];
  waiting_list: WaitingListItem[];
  heatmap: HeatmapItem[];
};